crates/egui/src/widgets/text_edit/builder.rs

```diff
@@ -746,6 +746,10 @@ impl<'t> TextEdit<'t> {
             }
         }
 
+        if state.ime_enabled && response.lost_focus() {
```

Comment:
Couldn`t you just leave the `state.ime_enabled` ? Because if its true its getting set to false and else it is already false, so to me it seems like an unnecessary condition

Reply:
> Couldn`t you just leave the `state.ime_enabled` ? Because if its true its getting set to false and else it is already false, so to me it seems like an unnecessary condition

When focus is moved elsewhere, you must set `ime_enabled` to 'false`, otherwise the IME will have problems when focus returns.

Reply:
sure the idea was the following
```rust
if response.lost_focus() {
state.ime_enabled = false;
}
```

Reply:
> sure the idea was the following
> 
> ```rust
> if response.lost_focus() {
> state.ime_enabled = false;
> }
> ```

Yes, I thought about this too. If ime_enabled is not 'true`, there is no need to check lost_focus() , so there is a difference in speed and readability. Let's wait for emilk's opinion on which one is better.

thank you.

Reply:
Ok but if so it would be good to add a comment there anyway